eng/UpdateAzureSdkForNet.ps1

```diff
@@ -1,20 +1,18 @@
-param($AutorestCSharpVersion, $CadlEmitterVersion, $SdkRepoRoot)
+param([string]$AutorestCSharpVersion, [string]$CadlEmitterVersion, [string]$SdkRepoRoot, [string[]]$ServiceDirectoryFilters = @("*"))
 
-$SdkRepoRoot = Resolve-Path $SdkRepoRoot
-
-Write-Host "Running Autorest.CSharp($AutorestCSharpVersion) and Cadl Emitter($CadlEmitterVersion) under $SdkRepoRoot"
+Import-Module "$PSScriptRoot\UpdateGeneratorMetadata.psm1" -DisableNameChecking -Force;
 
-$PackagesProps = "$SdkRepoRoot\eng\Packages.Data.props"
-(Get-Content -Raw $PackagesProps) -replace`
-    '<PackageReference Update="Microsoft.Azure.AutoRest.CSharp" Version=".*?" />',
-    "<PackageReference Update=`"Microsoft.Azure.AutoRest.CSharp`" Version=`"$AutorestCSharpVersion`" PrivateAssets=`"All`" />" | `
-    Set-Content $PackagesProps -NoNewline
+$SdkRepoRoot = Resolve-Path $SdkRepoRoot
 
-$CadlEmitterProps = "$SdkRepoRoot\eng\emitter-package.json"
-(Get-Content -Raw $CadlEmitterProps) -replace`
-    '"@azure-tools/cadl-csharp": ".*?"',
-    "`"@azure-tools/cadl-csharp`": `"$CadlEmitterVersion`"" | `
-    Set-Content $CadlEmitterProps -NoNewline
+UpdateGeneratorMetaData $AutorestCSharpVersion $CadlEmitterVersion $SdkRepoRoot
 
-dotnet msbuild /restore /t:GenerateCode "$SdkRepoRoot\eng\service.proj"
-dotnet msbuild /restore /t:GenerateTests "$SdkRepoRoot\eng\service.proj"
+Write-Host "Generating Azure SDK Codes..."
+foreach ($filter in $ServiceDirectoryFilters) {
+    Write-Host 'Generating projects under service directory ' -ForegroundColor Green -NoNewline
+    Write-Host "$filter" -ForegroundColor Yellow
+    dotnet msbuild /restore /t:GenerateCode /p:ServiceDirectory=$filter "$SdkRepoRoot\eng\service.proj"
+    if ($LastExitCode -ne 0) {
+        Write-Error 'Generation error'
+        exit 1
+    }
+}
```

Comment:
This part is identical with the script `eng/UpdateAzureSdkCodes.ps1`, should we call that script instead here?